# Patch from Jeroen van Wolffelaar <jeroen@wolffelaar.nl> to make
# arptables --proto-type also accept hexadecimal inputs (ethernet protocol
# numbers are often specfied in hex, not decimal), using standard strtol()
# behaviour (hex iff starts with 0x).

diff -urNad arptables-0.0.3.3~/arptables.c arptables-0.0.3.3/arptables.c
--- arptables-0.0.3.3~/arptables.c	2007-08-19 15:04:51.000000000 +0200
+++ arptables-0.0.3.3/arptables.c	2008-05-08 19:16:43.000000000 +0200
@@ -2039,7 +2039,7 @@
 			check_inverse(optarg, &invert, &optind, argc);
 			set_option(&options, OPT_P_TYPE, &fw.arp.invflags,
 				   invert);
-			if (get16_and_mask(argv[optind - 1], &fw.arp.arpro, &fw.arp.arpro_mask, 10)) {
+			if (get16_and_mask(argv[optind - 1], &fw.arp.arpro, &fw.arp.arpro_mask, 0)) {
 				if (strcasecmp(argv[optind-1], "ipv4"))
 					exit_error(PARAMETER_PROBLEM, "Problem with specified protocol type");
 				fw.arp.arpro = htons(0x800);
